src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,80 @@
+package core.basesyntax;
+
+import core.basesyntax.models.*;
+
+public class Main {
+    public static void main(String[] agrs) {
+        FigureSupplier figureSupplier = new FigureSupplier();
+        Figure[] figures = new Figure[6];
+
+        for (int i = 0; i < 3; i++) {
+            figures[i] = figureSupplier.getRandomFigure();
+        }
+
+        for (int i = 3; i < 6; i++) {
+            figures[i] = figureSupplier.getDefaultFigure();
+        }
+
+        for (Figure figure : figures) {
+            System.out.println("Figure: " + getFigureName(figure)

```

Comment:
The method `getFigureName` is not the best approach for obtaining the name of the figure. Instead of using `instanceof` checks, consider adding a `getName` method to the `Figure` class and override it in each concrete figure class. This is more in line with object-oriented design principles.